### ISSUE TITLE
object_length_linter and object_name_linter crash when lint is called with empty filename

### ISSUE THREAD

**Issue by jbodart-argenx** [jimhester/lintr#3032]
## Description

When `lint('', text = ...)` is called with an empty string as `filename`, `object_length_linter` and `object_name_linter` crash on Windows with:

```
Linter `linter()` failed in '':
Caused by error in `normalizePath()`:
! path[1]="": The system cannot find the path specified
```

This happens because both linters call `find_package(source_expression$filename)`, which calls `normalize_path('', mustWork = TRUE)`, which fails on Windows when the filename is empty.

This is triggered in practice when the R language server (`languageserver`) calls `lintr::lint(path, text = content)` and the path is empty (e.g., for `untitled:` or `vscode-userdata:` URI schemes where `path_from_uri()` returns `""`).

## Minimal Reprex

```r
library(lintr)
txt <- c("x <- 1", "camelCase <- 2")

# Works fine:
lint(text = txt)

# Crashes (object_length_linter calls find_package -> normalizePath(""))
lint("", text = txt)
```

**Expected:** No crash; linters should handle empty filename gracefully when `text=` is provided.

**Actual:** Error in `normalizePath(path[1])` from `find_package()`.

## Root Cause

In `R/get_source_expressions.R`, `find_package()` calls `normalize_path(filename, ...)` without checking if `filename` is empty or `nzchar()`.

Both `object_length_linter` and `object_name_linter` pass `source_expression$filename` to `find_package()` unconditionally. When `lint()` receives `filename = ""` and `text = ...`, it sets `inline_data = TRUE` so `normalize_path("", mustWork = FALSE)` succeeds in `lint()`, but `find_package()` still calls `normalize_path("", mustWork = TRUE)`.

## Suggested Fix

Guard `find_package()` against empty filename at the top of the function:

```r
find_package <- function(filename, ...) {
  if (!nzchar(filename)) return(NULL)
  # ... rest of function
}
```

This is also related to #2847; the `inline_data` / `parse_settings` interaction means that `lint("", text=...)` defaults to `parse_settings = FALSE`, so even if `.lintr` disables these linters, the config is not read.

## Session Info

- lintr 3.3.0.1
- R 4.5.0 on Windows 10
- languageserver 0.3.16
